parse/arrival-or-departure.js

```diff
@@ -27,11 +27,19 @@ const createParseArrOrDep = (prefix) => {
 			stop: d.stbStop.location || null,
 			...profile.parseWhen(ctx, d.date, tPlanned, tPrognosed, tzOffset, cancelled),
 			...profile.parsePlatform(ctx, plPlanned, plPrognosed, cancelled),
-			// todo: for arrivals, this is the *origin*, not the *direction*
-			direction: prefix === DEPARTURE && d.dirTxt && profile.parseStationName(ctx, d.dirTxt) || null,
-			provenance: prefix === ARRIVAL && d.dirTxt && profile.parseStationName(ctx, d.dirTxt) || null,
+			headsign: d.dirTxt && profile.parseStationName(ctx, d.dirTxt) || null,
 			line: d.line || null,
-			remarks: []
+			remarks: [],
+			origin: null,
+			destination: null
+		}
+
+		if (prefix === DEPARTURE && Array.isArray(d.prodL) && d.prodL[0].tLocX) {
+			res.destination = profile.parseLocation(ctx, ctx.res.common.locL[d.prodL[0].tLocX])
```

Comment:
Wow! You have discovered an aspect of the HAFAS mgate.exe response format that I didn't know about, even though I have worked with/against it for quite a while now. 🎉 

We should double-check if this actually *always* is the destination/origin (whatever that exactly means, there are lots of nuanced cases where the notion of a terminus stop is not so clear). **As `tLocX` usually specifies the stop that a certain *product* runs to, we need to check what happens with trips that change their *product* along their course, e.g. in-seat-transfer or shuttle buses.**

If `tLocX` turns out to be consistent with what one would consider the destination of a departing trip, let's merge it!